### PR TITLE
ISPN-9722 Cache v2 Resource

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-9772

Also includes a refactor in the REST documentation, separating the old and the new (v2) API

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+   private CompletionStage<RestResponse> removeCache(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String cacheName = request.variables().get("cacheName");
+      Cache<?, ?> cache = restCacheManager.getCache(cacheName, request.getSubject());
+      if (cache == null) {
+         responseBuilder.status(HttpResponseStatus.NOT_FOUND);
+         return CompletableFuture.completedFuture(responseBuilder.build());
+      }
+      return CompletableFuture.supplyAsync(() -> {
```
Comment (ryanemerson) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java]: Shouldn't we be passing an executor here? Was this missed by will's checkstyle PR?

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      } else if (sourceType.match(MediaType.APPLICATION_XML)) {
+         ConfigurationBuilderHolder builderHolder = PARSER_REGISTRY.parse(new String(bytes, UTF_8));
+         cfgBuilder = builderHolder.getCurrentConfigurationBuilder();
+      } else {
+         responseBuilder.status(HttpResponseStatus.UNSUPPORTED_MEDIA_TYPE);
+         return CompletableFuture.completedFuture(responseBuilder.build());
+      }
+
+      ConfigurationBuilder finalCfgBuilder = cfgBuilder;
+      return CompletableFuture.supplyAsync(() -> {
```
Comment (ryanemerson) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java]: Executor?

**Comment by ryanemerson** [comment]
@oraNod Can you review the docs please

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java (hunk 1 above)
Comment (gustavonalle) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java]: You're right, it was a bad rebase, will fix it

**Comment by gustavonalle** [comment]
@ryanemerson fixed

**Comment by gustavonalle** [comment]
@ryanemerson now it includes @oraNod reviews

**Pull request merged**